### DISCUSSION TITLE
LSP Auto complete with parenthesis

### DISCUSSION THREAD

**Discussion by Proful** [helix-editor/helix#5999] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/354596/219001844-77d12bbe-e9a4-4066-8834-02cd7c078cc5.png)

Expected (after pressing enter):-
```<caption>{k.toUpperCase()}</caption>```

Actual:-
```<caption>{k.toUpperCase}</caption>```

Any idea?

**Comment by the-mikedavis** [comment, marked as answer]
This behavior comes from the language server - it's sending just the `toUpperCase` and not the parentheses. This is very consistent behavior across language servers from what I've seen. Language servers typically complete paretheses if snippet support is enabled (#5864) because the server can also complete the function/method arguments with snippets.

Completing the parentheses in this case would probably be ok but it would only work for functions/methods with no arguments and it might be awkward if you have functions that can accept different numbers of arguments (for example, providing an optional argument in a language like Elixir).

Also see https://github.com/helix-editor/helix/issues/2678 which is different but has some overlapping discussion.
